UI/window-basic-main.cpp

```diff
@@ -7618,13 +7618,10 @@ void OBSBasic::on_actionShowSettingsFolder_triggered()
 
 void OBSBasic::on_actionShowProfileFolder_triggered()
 {
-	std::string userProfilePath;
-	userProfilePath.reserve(App()->userProfilesLocation.u8string().size() + OBSProfilePath.size());
-	userProfilePath.append(App()->userProfilesLocation.u8string()).append(OBSProfilePath);
-
-	const QString userProfileLocation = QString::fromStdString(userProfilePath);
+	const OBSProfile &currentProfile = GetCurrentProfile();
+	const QString currentProfileLocation = QString::fromStdString(currentProfile.path.u8string());
```

Comment:
Even though at first I too subscribed to "const all the things", I have learned that it does have unfortunate possible side-effects in C++:

Mainly that the move constructor (which moves elements of an old reference into a new) will fall back to a copy constructor, because the move constructor usually has to change the internal data pointers to `nullptr`, which it cannot do if the variable is declared `const`.

So if `QUrl::fromLocalFile` would be using a move constructor, it would exhibit that same fallback behaviour. The performance hit might be negligible here, but it taught me to be careful about making variables passed to other functions (or used as return values) `const` by default.

Reply:
I simply reused the same code from the function directly above, `on_actionShowSettingsFolder_triggered`. I've now adjusted my code based on your recommendation.